region/brinstar/green/Green Hill Zone.json

```diff
@@ -323,10 +348,30 @@
           "HiJump",
           "h_canFly",
           "canWalljump",
-          "canUseFrozenEnemies"
+          "canUseFrozenEnemies",
+          "canTrickySpringBallJump"
         ]}
       ]
     },
+    {
+      "link": [1, 4],
+      "name": "Shinespark Down Grab",
+      "requires": [
+        "Morph",
+        "canDownGrab",
+        "canShinechargeMovementComplex",
+        {"canShineCharge": {
+          "usedTiles": 18,
+          "openEnd": 1
+        }},
+        {"shinespark": {"frames": 15}}
+      ],
+      "note": "Spark flush against the right wall (diagonal helps) then quickly down grab after to barely get into the pipe.",
+      "devNote": [
+        "This only gets one pixel into the pipe, so a wiggle doesn't seem possible.",
```

Comment:
I checked it out and you can stand up in the pipe by doing a down-back instead of down-grab. The inputs are you start holding down any time during the crash animation, and then begin also holding left when it finishes (as you start to fall). So with that method, it's possible to do any of tunnel crawling, turnaround aim-canceling, or X-ray wiggling.